### PR TITLE
Add a new Communication app that manages all of Oscar's communications.

### PR DESCRIPTION
**Pull request by solarissmoke**
This is a rebase of #2591, which has been reviewed. @craigloftus could you please have a quick final scan of this before we merge?

### CODE REVIEW

File: src/oscar/apps/communication/config.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from django.apps import AppConfig
+from django.utils.translation import ugettext_lazy as _
```
Comment (craigloftus) [inline src/oscar/apps/communication/config.py]: Should be `gettext_lazy`

File: src/oscar/apps/customer/forms.py [hunk 2]
```diff
@@ -38,7 +37,6 @@ class PasswordResetForm(auth_forms.PasswordResetForm):
     """
     This form takes the same structure as its parent from :py:mod:`django.contrib.auth`
     """
-    communication_type_code = "PASSWORD_RESET"
 
     def save(self, domain_override=None, use_https=False, request=None,
```
Comment (craigloftus) [inline src/oscar/apps/customer/forms.py]: Can drop `use_https` as a named parameter.

File: tests/integration/templatetags/test_oscar_thumbnail.py [hunk 3]
```diff
@@ -10,8 +10,8 @@
 
 
 class OscarThumbnailMixin:
-    # Next values will be different `sorl-thumbnail` and `easy-thumbnails`.
-    crop_value = size_string_mapping = None
+    # Next value will be different for `sorl-thumbnail` and `easy-thumbnails`.
+    crop_value = None
```
Comment (craigloftus) [inline tests/integration/templatetags/test_oscar_thumbnail.py]: Not sure what this is doing in this PR?

**Comment by solarissmoke** [comment]
@craigloftus I've addressed your comments. Also regenerated migrations to as few as possible.

**Comment by solarissmoke** [comment]
Great job on this @sasha0 @samitnuk .

**Pull request merged**